mne/stats/regression.py

```diff
@@ -246,6 +248,16 @@ def linear_regression_raw(raw, events, event_id=None, tmin=-.1, tmax=1,
     """
 
     if isinstance(solver, string_types):
+        if solver == 'cholesky':
+            try:
+                from sklearn.linear_model import Ridge
+                def solver(X, y):
+                    return Ridge(fit_intercept=False, copy_X=False, alpha=0.,
+                                 solver='cholesky').fit(X, y.T).coef_
+            except ImportError:
+                warn("Can't import from scikit-learn. Falling back to 'pinv'.")
```

Comment:
...and instead of `try/except` we usually use a hopefully cleaner `sklearn` version check. If you look in `mne/cov.py` you should see some functions that simplify the checks.